client/src/main/java/com/paypal/selion/internal/platform/grid/browsercapabilities/CapabilitiesHelper.java

```diff
@@ -56,35 +63,140 @@ private CapabilitiesHelper() {
+            String providerString = (String) eachProvider;
+            if (StringUtils.isBlank(providerString)) {
+                continue;
+            }
+
             try {
-                if (DefaultCapabilitiesBuilder.class.isAssignableFrom(Class.forName(eachProvider))) {
-                    capsObjects.add(((DefaultCapabilitiesBuilder) Class.forName(eachProvider).newInstance())
-                            .createCapabilities());
+                Class<?> provider = Class.forName(providerString);
+                if (DefaultCapabilitiesBuilder.class.isAssignableFrom(provider)) {
+                    capsObjects.add(retrieveCustomCapabilities((Class<? extends DefaultCapabilitiesBuilder>) provider));
                 } else {
-                    logger.info("Skipping " + eachProvider + " because it is not a subclass of "
+                    logger.info("Skipping " + providerString + " because it is not a subclass of "
                             + DefaultCapabilitiesBuilder.class.getCanonicalName());
                 }
-            } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+            } catch (ClassNotFoundException e) {
                 // Throw an Un-checked exception and let the user know that the custom capabilities that they provided
-                // us with has problems.
-                // Doing this will prevent their tests from running under the assumption that the capabilities they
-                // provided was fine.
+                // us with has problems.Doing this will prevent their tests from running under the assumption that the
+                // capabilities they provided were fine.
                 throw new IllegalStateException(e);
             }
         }
         logger.exiting(capsObjects);
         return capsObjects;
     }
 
+    /**
+     * Parse capabilities from an array of String which uses the "name:value" format
+     * 
+     * @param capabilitiesArray
+     *            the capabilities to parse
+     * @return the parsed capabilities as a {@link DesiredCapabilities} object
+     */
+    public static DesiredCapabilities retrieveCustomCapabilities(String[] capabilitiesArray) {
+        logger.entering((Object[]) capabilitiesArray);
+
+        DesiredCapabilities caps = new DesiredCapabilities();
+        if (capabilitiesArray.length != 0) {
+            Map<String, Object> capabilityMap = parseIntoCapabilities(capabilitiesArray);
+            // We found some capabilities. Lets merge them.
+            caps = new DesiredCapabilities(capabilityMap);
+        }
+
+        logger.exiting(caps);
+        return caps;
+    }
+
+    /**
+     * Acquire capabilities from the TestNG {@link InvokedMethodInformation}
+     * 
+     * @param methodInfo
+     *            the TestNG {@link InvokedMethodInformation}
+     * @return the provided {@link DesiredCapabilities} which are associated with the {@link InvokedMethodInformation}
+     */
+    @Deprecated
```

Comment:
Because ExtendedConfig (which this method uses) is Deprecated. Once removed, this method will also be removed. I want to EOL the approach as an option for specifying additional desired capabilities